### DISCUSSION TITLE
The parameter "nelmio_security.csp" must be defined" error in Symfony 5.4 with NelmioSecurityBundle

### DISCUSSION THREAD

**Discussion by meherarbi** [symfony/symfony#49971] [Q&A]
Hello,

I'm facing an issue with the NelmioSecurityBundle in my Symfony application. I have configured the CSP settings in the nelmio_security.yaml file but I keep getting the following error:

"The parameter "nelmio_security.csp" must be defined."

I have tried clearing the cache, double-checked my configuration, and even reinstalled the bundle, but the error persists. Can you please help me resolve this issue?

Thank you.

`` nelmio_security:
    # signs/verifies all cookies
    signed_cookie:
        names: ['*']
    # prevents framing of the entire site
    clickjacking:
        paths:
            '^/.*': DENY
        hosts:
            - '^monsite\.com$'
    csp:
        enabled: true
        hosts: []
        enforce:
            default-src: ['self']
            script-src: ['self', 'unsafe-inline', 'https://ajax.googleapis.com', 'https://www.google-analytics.com']
            style-src: ['self', 'unsafe-inline']
            img-src: ['self', 'data:', 'https://www.google-analytics.com']
            font-src: ['self', 'data:']
            connect-src: ['self', 'https://www.google-analytics.com']
        report:
            report-uri: "/csp/report"
            script-src:
                - 'self'
    referrer_policy:
        enabled: true
        policies:
            default: 'no-referrer'
``

**Comment by elevado** [comment]
Can you please show the nelmio_security.yaml here? If necessary, mask values in the file if they are sensitive.

Also have a look at the part of the documentation: https://symfony.com/bundles/NelmioSecurityBundle/current/index.html#maximum-security-configuration

**Reply by meherarbi** [reply]
```yaml
nelmio_security:
    # signs/verifies all cookies
    signed_cookie:
        names: ['*']
    # prevents framing of the entire site
    clickjacking:
        paths:
            '^/.*': DENY
        hosts:
            - '^monsite\.com$'
    csp:
        enabled: true
        hosts: []
        enforce:
            default-src: ['self']
            script-src: ['self', 'unsafe-inline', 'https://ajax.googleapis.com', 'https://www.google-analytics.com']
            style-src: ['self', 'unsafe-inline']
            img-src: ['self', 'data:', 'https://www.google-analytics.com']
            font-src: ['self', 'data:']
            connect-src: ['self', 'https://www.google-analytics.com']
        report:
            report-uri: "/csp/report"
            script-src:
                - 'self'
    referrer_policy:
        enabled: true
        policies:
            default: 'no-referrer'
```

**Reply by elevado** [reply]
It would have been better if you had added the config formatted as code here. But I see `csp` there.

You have to debug in the bundle how the configuration gets there. The error message _"The parameter "nelmio_security.csp" must be defined."_ means that the property `csp` does not exist.

**Reply by stof** [reply]
You would have to search in your project for usages of `nelmio_security.csp` as a DI parameter (so search first for `%nelmio_security.csp%` as most usages of parameters are done through this notation).
What you pasted is the semantic configuration of the bundle, and there is no guarantee that the bundle creates such a parameter based on this configuration.

Also, the stack trace of the exception you get could help debugging the issue as well (if you get this in the console, run your command in verbose mode to display the stack trace)
